File: 算法训练营/03-数组链表跳表/206/迭代法.go
/*
 * @lc app=leetcode.cn id=206 lang=golang
 *
 * [206] 反转链表
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func reverseList(head *ListNode) *ListNode {
	root := new(ListNode)
	for head != nil {
		tmp := root.Next 
		root.Next = head
		head = head.Next
		root.Next.Next = tmp // 最后赋值末尾, 顺序不能乱
	}
	return root.Next
}
// @lc code=end

File: LeetCode/0206/迭代法.py
#
# @lc app=leetcode.cn id=206 lang=python3
#
# [206] 反转链表
#

# @lc code=start
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        pre, cur, nxt = None, head, None
        while cur:
            nxt = cur.next
            cur.next = pre
            pre, cur = cur, nxt
        return pre
# @lc code=end